Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -56,25 +60,25 @@ namespace WebCore {
 
 WTF_MAKE_ISO_ALLOCATED_IMPL(CookieStore);
 
-Ref<CookieStore> CookieStore::create(Document* document)
+Ref<CookieStore> CookieStore::create(ScriptExecutionContext* context)
 {
-    auto cookieStore = adoptRef(*new CookieStore(document));
+    auto cookieStore = adoptRef(*new CookieStore(context));
     cookieStore->suspendIfNeeded();
     return cookieStore;
 }
 
-CookieStore::CookieStore(Document* document)
-    : ActiveDOMObject(document)
+CookieStore::CookieStore(ScriptExecutionContext* context)
+    : ActiveDOMObject(context)
 {
-    if (!document)
-        return;
-
-    auto* page = document->page();
-    if (!page)
-        return;
+    if (context->isDocument()) {
```

Comment:
`is<Document>(context)`.

Your current code is not safe as context might be null.